test/CMakeLists.txt

```diff
@@ -74,6 +74,7 @@ if (NOT (MSVC AND BUILD_SHARED_LIBS))
 endif ()
 add_fmt_test(ostream-test)
 add_fmt_test(compile-test)
+add_fmt_test(compile-fp-test HEADER_ONLY)
```

Comment:
I suggest adding
```
if (MSVC)
  target_compile_options(compile-fp-test PRIVATE /utf-8 /Zc:__cplusplus
                         /Zc:externConstexpr /Zc:inline)
endif ()
```
to catch portability problems like 'module-test' does.

Reply:
done, but only with `/Zc:__cplusplus`, since other options seem unrelated for this, but I can be wrong here

Reply:
You were asking for full conformance to the standard from Microsoft, disregarding customer demands (i.e. not following stated business policies). This is what you need to get that in recent language modes. In particular when it comes to anything `constexpr` related.

BTW: what is the reason to demand `HEADER_ONLY`? This sounds a little like a portability or conformance problem to me.

Reply:
I ran my tests again with your updated PR, using the compiler flags as suggested for correct behaviour of `constexpr`:

msvc 16.10 (i.e. VS2019 latest): ✅
msvc 16.11-pre3 (i.e. VS2019 Preview): ✅
msvc 17.0-pre2 (i.e. VS2022 Preview): ✅

All 20 tests compile and pass.
The library also compiles in all modes at /W4 without warnings.

Congratulations!

But I'm not happy about this `HEADER_ONLY` situation.

Reply:
I'm not happy about `HEADER_ONLY` too, but I don't know how to implement it in the better way.
Also, I just hope it would not introduce some core issues, like the recently discovered issue [when mixing format.h with and without ostream.h in different cpp files caused segfault](https://github.com/fmtlib/fmt/issues/2357) _(which is, IMHO, unsolvable without changing some core ideas)_.